src/config.js

```diff
@@ -33,27 +31,62 @@ export function applyConfigToArgv({
         `specified in camel case: "${camelCase(option)}"`);
     }
 
-    if (option === commandExecuted) {
+    if (typeof options[option] === 'object' &&
+      typeof configObject[option] === 'object') {
+      // Descend into the nested configuration for a sub-command.
       newArgv = applyConfigToArgv({
         argv,
-        configObject: configObject[commandExecuted],
-        defaultValues: defaultValues[commandExecuted],
+        configObject: configObject[option],
+        options: options[option],
         configFileName});
       continue;
     }
 
+    const decamelizedOptName = decamelize(option, '-');
+    let expectedType;
+    if (options[decamelizedOptName]) {
+      if (options[decamelizedOptName].type === undefined) {
```

Comment:
This doesn't work for some reason. It keeps trying to read `type` even if the object is `undefined`. I'm not sure why.